src/ontoweaver/make_value.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import abc
+
+from . import errormanager
+from . import exceptions
+
+logger = logging.getLogger("ontoweaver")
+class ValueMaker(errormanager.ErrorManager, metaclass=abc.ABCMeta):
+    """
+    Interface for value makers that are used to define the extraction logic for each of the transformer classes.
+    """
+    def __init__(self, raise_errors: bool = True):
+        """
+        Initialize the ValueMaker with the option to raise errors.
+
+        Args:
+            raise_errors (bool): if True, will raise an exception when an error is encountered, else, will log the error and try to proceed.
+        """
+
+        super().__init__(raise_errors)
+
+    def check_attributes(self, **kwargs):
+        """
+        Check if all attributes expected in the __init__ are set. If not, raise an error.
+        """
+
+        expected_att = self.__dict__
+        self.__dict__.update(**kwargs)
+
+        expected_keys = {k: v for k, v in expected_att.items() if not k.startswith('__')}
+        for k, v in expected_keys.items():
+            if v is None:
+                self.error(f"Attribute {k} not set.", section=f"{self.__class__.__name__}.check", exception=exceptions.TransformerDataError)
+
+    @abc.abstractmethod
+    def select(self, columns, row, i):
```

Comment:
Let's follow the same naming convention than for the LabelMaker class:
```suggestion
    def call(self, columns, row, i):
```

Reply:
Or maybe even use the `__call__` interface, since we should not need `check_attributes`.

Reply:
What kind of errors?

Reply:
I was wrong - the way it should have been done is to instantiate the class once all the parameters are already passed to the `__init__`. This is how it is now, with no use of the parameter checking functions.